projecttacoma/deqm-test-server-frontend
# operation menus

## Summary
Renders evaluate measure and care gaps buttons on specific resource pages to redirect to evaluate or care gaps page for a given measure. The corresponding fields are then pre-populated with data from the original resource page.  

## New behavior
Patient, Practitioner and Group resources now contain an evaluate measure button, and Patient, Practitioner, and Organization resources now contain a care gaps button. When clicked, these buttons allow the user to select a measure from a dropdown menu and reroute to that measure's evaluate or care gaps page. The patient, practitioner, group, and organization field is populated based on the original resource.  

## Code changes
- **components/ResourceMenu.tsx**, menu component added that displays clickable measure resources that navigate to the specified page 
- ** pages/[resourceType]/[id]/care-gaps.tsx**, added functionality to retrieve patient, organization, and practitioner information from the URL to prepopulate select fields. 
- ** pages/[resourceType]/[id]/evaluate-measure.tsx**, added functionality to retrieve patient, group, and practitioner information from the URL to prepopulate select fields.
- ** pages/[resourceType]/[id]/index.tsx**, renders care gap menu buttons on patient, practitioner, and organization resources and evaluate measure buttons on patient, practitioner, and group. 
- - **__tests__/pages/resource/id/index.test.tsx**, unit tests added
- - **__tests__/pages/resource/id/evaluate.test.tsx**, unit tests added
- - **__tests__/pages/resource/id/care-gaps.test.tsx**, unit tests added
## Testing Guidance
- See the README.md for first time setup instructions.
#### Testing in browser: 
To start the frontend: `npm run dev` then navigate to http://localhost:3001 in a browser
- Navigate to any patient resource ID page. Click on the evaluate measure button and select any measure from the drop down. You should be redirected to that measure’s evaluate measure page and the patient field should be pre-populated with the patient data. Click calculate and verify that the patient appears in the report. 
- Repeat this process with the care gaps button on the patient page. 
- Navigate to any participant resource ID page and repeat this process with the care gaps and evaluate measure buttons 
- Navigate to any group resource ID page and repeat this process with the evaluate measure button. Note that the care gaps button should not appear
- Navigate to any organization resource ID page and repeat this process with the care gaps button. Note that the evaluate measure button should not appear
#### Unit testing: 
- Run the unit tests: `npm run test`
